packages/material-ui-styles/src/makeStyles/makeStyles.d.ts

```diff
@@ -70,7 +69,7 @@ export type StylesHook<S extends Styles<any, any>> = StylesRequireProps<S> exten
   ? (props?: any) => ClassNameMap<ClassKeyOfStyles<S>>
   : (props: PropsOfStyles<S>) => ClassNameMap<ClassKeyOfStyles<S>>;
 
-export default function makeStyles<S extends Styles<any, any>>(
-  styles: S,
-  options?: WithStylesOptions<ThemeOfStyles<S>>,
-): StylesHook<S>;
+export default function makeStyles<C extends string, P extends {}, T = unknown>(
```

Comment:
If we change this up a bit to 
```suggestion
export default function makeStyles<T = unknown, P extends {} = {}, C extends string = string>(
```

Can we still use `makeStyles<MyTheme>(theme => ({ root: {} }))` and TypeScript infers the proper `ClassKey`? Allows a different flavor of TypeScript depending on where people want to put their types.

Oh and while we're at it: I'll switched to descriptive type parameter names: `Styles<T, P, C>` is not very descriptive. Could you write those out?

Reply:
I can swap the order and change to descriptive type parameters. Unfortunately if you only provide Theme, then the other two won't infer, but will just use the default parameters. There is a [proposal](https://github.com/Microsoft/TypeScript/issues/26242) to allow partial inferrence.

Reply:
Right. That's the one thing I still miss from `flow`: `makeStyles<*, Props, *>`. Thanks for linking the issue. Didn't know that was on their radar.

No need to change the order if `Styles` uses the same. Would like to see descriptive names though.

Reply:
Updated order and names to match `Styles`.